wave_benchmark/src/trajectory_compare.cpp

```diff
@@ -0,0 +1,44 @@
+#include "wave/benchmark/trajectory_compare.hpp"
+
+namespace wave {
+
+void TrajectoryCompare::reset() {
+    this->ground_truth.clear();
+    this->measurements.clear();
+    this->error.clear();
+}
+
+void TrajectoryCompare::push_measurement(Pose pose, const TimeType time) {
+    measurements.insert(PoseMeasurement(time, this->measurement_key, pose));
+}
+
+void TrajectoryCompare::push_truth(Pose pose, const TimeType time) {
+    ground_truth.insert(PoseMeasurement(time, this->ground_truth_key, pose));
+}
+
+void TrajectoryCompare::calculate_error() {
```

Comment:
This function looks good. In fact, it should be more prominent. It is the meat of the TrajectoryCompare class. The whole interface should be designed around it. 

 When someone wants to calculate error, what do they typically have? Two sequences of measurements.

They already have two sequences of measurements. Why should they have to learn how to feed the measurements-by-one into a `TrajectoryCompare` class, just so they can get at the juicy meat of `calculateError()`?

---

Here is the problem and solution in my view. You have two sequences of poses. You want to calculate the error. You want to print output to a csv file. What do you _really_ need?

```cpp
error_poses calculateError(estimate_poses, truth_poses);
void outputToCsv(poses, file path);
```

Do you really need the rest?